src/common/types/time.cpp

```diff
@@ -301,7 +304,7 @@ dtime_t Time::FromTime(int32_t hour, int32_t minute, int32_t second, int32_t mic
 
 bool Time::IsValidTime(int32_t hour, int32_t minute, int32_t second, int32_t microseconds) {
 	if (hour < 0 || hour >= 24) {
-		return false;
+		return (hour == 24) && (minute == 0) && (second == 0) && (microseconds == 0);
```

Comment:
Why are we limiting the TIME at 24 hours?
I thought we were allowing arbitrary time values in both TIME and INTERVAL?

Reply:
`TIME` is limited to `24:00:00` See https://www.postgresql.org/docs/current/datatype-datetime.html

Reply:
The issue I linked that I discovered in our Arrow scan is that Arrow does support > 24 hour time values, I don't know the limit but I got it to produce 36 hours, could be much higher.

Reply:
It does seem like Postgres hard caps at 24 hours though.
```
postgres=# select CAST('24:00:00' AS time);
   time   
----------
 24:00:00
(1 row)

postgres=# select CAST('36:00:00' AS time);
ERROR:  date/time field value out of range: "36:00:00"
LINE 1: select CAST('36:00:00' AS time);
                    ^
postgres=# select CAST('24:00:01' AS time);
ERROR:  date/time field value out of range: "24:00:01"
LINE 1: select CAST('24:00:01' AS time);
```

Does that mean we need to actively check scanned time values from Arrow to see if they break our accepted bounds?

Reply:
Ah. Well we used to be restricted to `23:59:59.999999` so it's a little better! But yes reading back could be an issue. But this is about PG compatibility.

Reply:
yes I think so.

> Does that mean we need to actively check scanned time values from Arrow to see if they break our accepted bounds?

Sadly I think so. Or maybe map them to intervals?